src/sentry/integrations/base.py

```diff
@@ -125,6 +125,53 @@ class IntegrationFeatures(Enum):
     DEPLOYMENT = "deployment"
 
 
+# Integration Types
+MESSAGING = "messaging"
+PROJECT_MANAGEMENT = "project_management"
+SOURCE_CODE_MANAGEMENT = "source_code_management"
+ON_CALL_SCHEDULING = "on_call_scheduling"
+
+
+class IntegrationProviderSlug(Enum):
+    SLACK = "slack"
+    DISCORD = "discord"
+    MSTeams = "msteams"
+    JIRA = "jira"
+    JIRA_SERVER = "jira_server"
+    AZURE_DEVOPS = "vsts"
+    GITHUB = "github"
+    GITHUB_ENTERPRISE = "github_enterprise"
+    GITLAB = "gitlab"
+    BITBUCKET = "bitbucket"
+    PAGERDUTY = "pagerduty"
+    OPSGENIE = "opsgenie"
+
+
+INTEGRATION_TYPE_TO_PROVIDER = {
+    MESSAGING: [
+        IntegrationProviderSlug.SLACK,
+        IntegrationProviderSlug.DISCORD,
+        IntegrationProviderSlug.MSTeams,
+    ],
+    PROJECT_MANAGEMENT: [
```

Comment:
Many (or all) of source code management integrations are also project management ones. At least Github and Azure are. @cathteng can confirm which ones for sure.

Reply:
are we grouping integrations by the main focus or by a particular feature? i would say the ticketing aspect of SCMs is a feature

Reply:
@sentaur-athena @cathteng I just grouped them based on our integration feature list

Reply:
@cathteng we can come back and update this list since it's not really used anywhere in code other than messaging. My thought is for example if we want to build onboarding and tell people we have these ticket management integrations we still want to return github in that list.

Reply:
ok